main.js

```diff
@@ -10,32 +10,38 @@ var currentPalette = [];
 var savedPalettes = [];
 
 // Event Listeners
-window.addEventListener('load', function() {
+window.addEventListener('load', function () {
 	generateRandomPalette();
 	updatePalette(currentPalette)
 });
 
-newPaletteBtn.addEventListener('click', function() {
+newPaletteBtn.addEventListener('click', function () {
 	generateRandomPalette();
 	updatePalette(currentPalette)
 });
 
-savePaletteBtn.addEventListener('click', function() {
+savePaletteBtn.addEventListener('click', function () {
 	savePalette();
 	displaySavedPalettes();
 	console.log(savedPalettes);
 });
 
-miniPaletteContainer.addEventListener('click', function(event) {
-	var target = event.target;
-	if (target.parentNode.classList.contains('delete-button')) {
-			var index = target.parentNode.dataset.index;
-			deleteSavedPalette(index);
-			console.log(savedPalettes);
+miniPaletteContainer.addEventListener('click', function (event) {
+	if (event.target.parentNode.classList.contains('delete-button')) {
+		var index = event.target.parentNode.dataset.index;
+		deleteSavedPalette(index);
+		console.log(savedPalettes);
+	} else if (event.target.className === 'mini-palette' || 'mini-box') {
+		var index = event.target.dataset.index;
+		console.log(index);
+		console.log(savedPalettes);
+		var savedPalette = savedPalettes[index];
+		updatePalette(savedPalette);//display the saved colors
+		currentPalette = savedPalette.slice();// now put those values into the currentPalette 
```

Comment:
This last line ensures that we are updating the data model and not just the display. Important key!